src/tree_node.h

```diff
@@ -20,6 +20,11 @@ class TreeNode {
 
   void setParent(TreeNode<V>* node);
   void addChild(TreeNode<V>* node);
+  // This method is updating a node and to set for example the right child of
+  // a binary tree without setting the left child.
+  // This method will fill upp the vector with enough cildren if the index is
+  // larger then the size of the vector.

```

Comment:
Would personally like a multiline comment instead (/* */) since it is multiple lines.